presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkMemoryBasedBroadcastDependency.java

```diff
@@ -60,19 +70,13 @@ public Broadcast<List<PrestoSparkSerializedPage>> executeBroadcast(JavaSparkCont
         // release memory retained by the RDD (splits and dependencies)
         broadcastDependency = null;
 
-        long compressedBroadcastSizeInBytes = broadcastValue.stream()
-                .mapToInt(page -> page.getBytes().length)
-                .sum();
-        long uncompressedBroadcastSizeInBytes = broadcastValue.stream()
-                .mapToInt(page -> page.getUncompressedSizeInBytes())
-                .sum();
-
         long maxBroadcastSizeInBytes = maxBroadcastSize.toBytes();
-
+        long compressedBroadcastSizeInBytes = isNativeExecution ? getNativeMaxSerializedBytes(broadcastValue, pagesSerde) : getCompressedBroadcastSizeInBytes(broadcastValue);
```

Comment:
Is there a reason we care about compressed size check ? We should be able to rely only on uncompressed size check for OOMing ..

Reply:
I just left the java behavior unchanged. At the same time, I can't see why it will ever be more than uncompressed size. I have updated the code to remove this check